src/analyses/base.ml

```diff
@@ -1997,6 +1997,20 @@ struct
     let st' = invalidate ~deep:false ~ctx (Analyses.ask_of_ctx ctx) gs st shallow_addrs in
     invalidate ~deep:true ~ctx (Analyses.ask_of_ctx ctx) gs st' deep_addrs
 
+  let check_free_of_non_heap_mem ctx special_fn ptr =
+    let points_to_set = ctx.ask (Queries.MayPointTo ptr) in
+    begin try
+        let exists_non_heap_var =
+          (* elements throws Unsupported if the points-to set is top *)
+          Queries.LS.elements points_to_set
+          |> List.map fst
+          |> List.exists (fun var -> not (ctx.ask (Queries.IsHeapVar var)))
+        in
+        if exists_non_heap_var then
+          M.warn ~category:(Behavior (Undefined InvalidMemoryDeallocation)) ~tags:[CWE 590] "Free of non-dynamically allocated memory in function %s for pointer %a" special_fn.vname d_exp ptr
+      with _ -> M.warn ~category:(Behavior (Undefined InvalidMemoryDeallocation)) ~tags:[CWE 590] "Points-to set for pointer %a in function %s is top. Potential free of non-dynamically allocated memory may occur" d_exp ptr special_fn.vname
```

Comment:
Catch-all exception handlers shouldn't be used. If `eval_rv_address` is used above, then the top pointer case can be handled without exceptions like elsewhere in base.

Reply:
Should also be fixed in https://github.com/goblint/analyzer/pull/1099/commits/99fb013038d3b6c333f236537668e3513d8faceb